src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -40,9 +40,20 @@
 
 /**
  * <p>
- * Checks that a class that has only private constructors and has no descendant
- * classes is declared as final.
+ * Check that identifies classes that can be declared as final are explicitly made final.
+ * The following is the type of classes that can be identified:
  * </p>
+ * <ol>
+ *   <li>
+ *       Private classes with implicit constructor.
+ *   </li>
+ *   <li>
+ *       Classes with all the type of modifier which contains only private constructor.
+ *   </li>
+ *   <li>
+ *       Class must not have any descendant class.
```

Comment:
It seems to me there is only 2 checks just like your code. `Class must not have any descendant class` is common between the 2.

Add a new paragraph after this list and say something like:
````
Classes are skipped over if:
  It is already be final,
  ...
````

Reply:
Done